test/code_analysis/warning_functions.bats

```diff
@@ -16,6 +16,8 @@
 	exceptions["dirname"]="sys.*"
 	exceptions["free"]="macros.h,verifytime.c,verifytime_main.c"
 	exceptions["calloc"]="macros.h,verifytime.c"
+	exceptions["strcmp"]="strings.h"
+	exceptions["strlen"]="strings.h"
```

Comment:
I don't see why using strlen and strcmp instead of the "safer" options strnlen and strncpm. And in any case, if it is really necessary to use strlen it is better to just add an exception for the file that requires them (delta.c) instead of changing them for all files. Changing them in strings.h is going to cause those unsafe functions to be used everywhere, not just in delta.c.

Reply:
As Arjan mentioned we are finding the strn* functions to be more problematic than the non-n versions, thus this change.

Reply:
strnlen and friends have the human put in a length

strlen/strcpy and friends, with FORTIFY_SOURCE, have hte compiler put in the length of the buffer automatically.. which is more accurate.

strncpy and friends also have the nasty behavior of not 0-terminating the strings

Reply:
I see. If that is the case, and the str_len and str_cmp macros are no longer needed, it may be better to just get rid of them instead of just using them as an alias for strlen and strcmp, and just remove those functions from https://github.com/clearlinux/swupd-client/blob/master/test/code_analysis/warning_functions.bats#L11. But that can probably done in a different PR.

Reply:
Yea, I was thinking about that but decided to keep the macro around for now. Who knows when things change and we need a new str* =P.